modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/SynonymTokenFilterFactory.java

```diff
@@ -62,8 +111,10 @@ public class SynonymTokenFilterFactory extends AbstractTokenFilterFactory {
                     + "Instead, insert a lowercase filter in the filter chain before the synonym_graph filter."
             );
         }
+
+        this.synonymsSource = SynonymsSource.fromSettings(settings);
         this.expand = settings.getAsBoolean("expand", true);
-        this.lenient = settings.getAsBoolean("lenient", false);
+        this.lenient = settings.getAsBoolean("lenient", true);
```

Comment:
Is this set `true` for all types of synonyms? From the title of PR it looks like it should `true` only for `synonyms_set` synonyms?

Reply:
Yes, we decided to make that change after discussion in this PR and I haven't updated the PR name yet